### DISCUSSION TITLE
Snapshot did not improve any syncing time.

### DISCUSSION THREAD

**Discussion by Ranjith980** [Mimetis/Dotmim.Sync#589] [Q&A]
Thank you for your work on Syncing process. Tremendous contribution. We are using sync Version:0.5.7 web service with the  sql server  and syncing it to sqlite Client. We have turned on the change tracking on the sql server.we have multiple  clients using an UWP application and using the syncing service. We are using indexing on both client and server. Also using GUIDs on sqlserver as primary key.
Reinitializing the database takes 16 minutes and using the snapshot takes 1:30 hours  with our database with 1million records currently and is expected to grow potentially. As its mentioned in the documentation on snapshots that “The idea is to reduce this time for initialization of new clients.” 
We could not decrease the time It takes to reinitialize. We are trying to analyze and get it working with decreasing  the  time. We are using the information on the documentation and we still see the time increasing with respect to reinitialization. Is there any detail document on how to correctly configure the snapshots?. Your inputs will be greatly helpful to work under this scenario.

**Comment by Mimetis** [comment]
How often are you creating the snapshot ?
Snapshots should not be created every time you are reinitializing a client. 
It should be created like every week or every 2/3 days, depending your need

Do you have filters applied to each new client ? If yes, are you creating a filtered snapshot for each client ?

Why are you reinitializing client ? 
Is it something you are doing often ?

**Reply by Ranjith980** [reply]
Thank you for the reply. We are testing the snapshot. 
We are scheduling the snapshot to test every two days as per your suggestion. 
We do not have filters applied. We need to reinitialize the client when there are schema changes. 
We do it as needed, but we do perform reinitialization.

**Reply by Mimetis** [reply]
when you are changing the schema, you need to recreate a new snapshot
